tools/validate_spec/validation_tool.py

```diff
@@ -0,0 +1,305 @@
+
+from rflx.error import RecordFluxError
+from rflx.identifier import ID
+from rflx.pyrflx import PyRFLX, TypeValue
+from rflx.pyrflx.typevalue import MessageValue
+
+
+def cli(argv: List[str]) -> Union[int, str]:
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="subcommand")
+    subparsers.required = True
+
+    parser_verify = subparsers.add_parser(
+        "verify", help="check specification against a set of known valid or invalid examples"
+    )
+
+    parser_verify.add_argument(
+        "-d-valid",
+        "--valid-directory",
+        type=Path,
+        help=("absolute path to directory with known valid examples"),
+        default=None,
+    )
+
+    parser_verify.add_argument(
+        "-d-invalid",
+        "--invalid-directory",
+        type=Path,
+        help=("absolute path to directory with known valid examples"),
+        default=None,
+    )
+
+    parser_verify.add_argument(
+        "-m",
+        "--message-identifier",
+        type=str,
+        help=("identifier of the top-level message: <PackageName>.<MessageName>"),
+        required=True,
+    )
+
+    parser_verify.add_argument(
+        "-o",
+        "--json-output",
+        type=Path,
+        help=("identifier of the top-level message: <PackageName>.<MessageName>"),
+        default=None,
+    )
+
+    parser_verify.add_argument(
+        "--abort-on-error",
+        action="store_true",
+        help=("abort with exitcode 1 if a message cannot be verified correctly"),
+    )
+
+    parser_verify.add_argument("-s", "--specification", type=Path, help=("specification file"))
+
+    parser_verify.set_defaults(func=verify)
+
+    args = parser.parse_args(argv[1:])
+    try:
+        args.func(args)
+    except RecordFluxError as e:
+        return f"{e}"
+
+    return 0
+
+
+def verify(args: argparse.Namespace) -> None:
+    validation_main(
+        args.message_identifier,
+        args.specification,
+        args.invalid_directory,
+        args.valid_directory,
+        args.json_output,
+        args.abort_on_error,
+    )
+
+
+def validation_main(
+    root_message_id: str,
+    path_spec: Path,
+    path_invalid: Optional[Path],
+    path_valid: Optional[Path],
+    full_output: Optional[Path],
+    abort_on_error: bool,
+) -> None:
+    # pylint: disable=too-many-locals
+    def get_message_to_validate(path: Path, is_vaild: bool) -> "Message":
+        if not path.is_file():
+            raise TypeError(f"{path} is not a regular file")
+        message_bytes = path.read_bytes()
+        message = Message(message_bytes, is_vaild, path.name)
+        return message
+
+    identifier = ID(root_message_id.split(".")).parts
+    assert len(identifier) == 2
+    msg_name = identifier[0]
+    package_name = identifier[1]
+    pdu_message = PyRFLX.from_specs([str(path_spec)])[msg_name][package_name]
+
+    path_generators: Dict[Path, Iterator[Path]] = {}
+    for path in [path_valid, path_invalid]:
```

Comment:
We should enforce this list to be non-empty, otherwise the tool will just return when neither a valid or an invalid samples directory is provided. I don't think this is what we want.

Reply:
Added a check in f057148